python/ray/serialization.py

```diff
@@ -476,6 +480,9 @@ def serialize(self, value):
             return RawSerializedObject(value)
 
         if self.worker.use_pickle:
+            assert hasattr(np.ndarray, "__reduce_ex__"), "Need "
+            "numpy >= 1.16.0 for use_pickle=True support. Use "
+            "ray.init(use_pickle=False) for older numpy versions."
```

Comment:
We should also say that this will degrade performance.

Reply:
The way it is at the moment, it just won't be possible to run the slow code path (either use_pickle=True or use_pickle=False will be fast, with the first one only supported on numpy >= 1.16.0).

Reply:
Ah yeah I just updated it to say the flag may be removed in the future. Want to encourage people to upgrade numpy and not use old serialization